modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleConfigurationConfigurator.java

```diff
@@ -37,7 +38,7 @@
  * only work if the MuleContext has not yet been started, otherwise the modifications 
  * will be ignored (and warnings logged).
  */
-public class MuleConfigurationConfigurator implements MuleContextAware, SmartFactoryBean
+public class MuleConfigurationConfigurator implements MuleContextAware, SmartFactoryBean, ObjectFactory
```

Comment:
makes sense to add a TODO so that this class stops implementing SmartFactoryBean in the near future?